parsec/core/fs/exceptions.py

```diff
@@ -1,57 +1,168 @@
 # Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
 
+import os
+import errno
 from parsec.core.types import EntryID
+from parsec.core.fs.utils import ntstatus
 
 
-class FSError(Exception):
-    pass
+# Base class for all file system errors
 
 
-class FSValidationError(FSError):
-    pass
+class FSError(OSError):
+    ERRNO = None
+    WINERROR = None
```

Comment:
Il sert à quelque chose lui ? je ne le vois pas utiliser dans le code (à part dans le `__init__` mais du coup il devrait être tout le temps à `None`...)

Reply:
Oui en fait les winerror sont différents des ntstatus et l'idée c'est que les `FSError` soient capable de fournir leur `self.errno`, `self.winerror` et `self.ntstatus`. Le problème c'est que j'avais pas une enum de winerror a disposition et on utilise pas les winerror pour l'instant mais à terme ce serait pas mal d'avoir:
```python
class FSNotADirectoryError(FSError, NotADirectoryError):
    ERRNO = errno.ENOTDIR
    NTSTATUS = ntstatus.STATUS_NOT_A_DIRECTORY
    WINERROR = winerror.ERROR_DIRECTORY
```

Reply:
Un exemple d'utilisation c'est qu'en windows les exceptions sont affichées sous la forme:

    NotADirectoryError: [WinError 267] The directory name is invalid: 'MANIFEST.in'

Donc ce serait pas mal d'afficher nos FSError de la meme maniere:

    FSNotADirectoryError: [WinError 267] The directory name is invalid: 'MANIFEST.in'